### PR TITLE
Add button to reset to default view

### PR DESCRIPTION
**Pull request by kalletlak**
# What? Why?
Fix https://github.com/cBioPortal/cbioportal/issues/6543

Screen Shot 2019-09-19 at 3 46 34 PM

### CODE REVIEW

File: src/pages/studyView/StudyViewPage.tsx [hunk 1]
```diff
@@ -329,8 +330,31 @@ export default class StudyViewPage extends React.Component<IStudyViewPageProps,
...
+                                        <Modal bsSize={'small'} show={this.showModal} onHide={() => { this.showModal = false; }} keyboard>
+                                            <Modal.Header closeButton>
+                                                <Modal.Title>Reset to Default view</Modal.Title>
+                                            </Modal.Header>
+                                            <Modal.Body>
+                                                <div>Are you sure you want to override the list and order of the selected charts
+                                                     with the default setting? This is irreversible.</div>
+                                            </Modal.Body>
+                                            <Modal.Footer>
+                                                <button className='btn btn-primary btn-sm'
```
Comment (zhx828) [inline src/pages/studyView/StudyViewPage.tsx]: User could exit the modal when decided not to restore by clicking the x icon on the top right. I think it also make sense to have a `No` button after `Yes` to naturally close the dialog.

File: src/pages/studyView/addChartButton/AddChartButton.tsx [hunk 2]
```diff
@@ -268,6 +269,20 @@ class AddChartTabs extends React.Component<IAddChartTabsProps, {}> {
                     />
                 </MSKTab>
             </MSKTabs>
+
+            <div style={{
```
Comment (zhx828) [inline src/pages/studyView/addChartButton/AddChartButton.tsx]: It could just be a span element, so you don't need to specify display flex

**Comment by jjgao** [comment]
- when the user is not logged in, maybe we don't need to "Default button"?
- Change the title of warning message to "Reset charts"
- Let's change the warning message to: "Please confirm that you would like to replace the current charts with the default list." and add "Confirm" and "Cancel" buttons.

**Comment by kalletlak** [comment]
@zhx828 @jjgao updated

File: src/pages/studyView/StudyViewPage.tsx [hunk 3]
```diff
@@ -73,6 +73,7 @@ export default class StudyViewPage extends React.Component<IStudyViewPageProps,
     private queryReaction:IReactionDisposer;
     @observable showCustomSelectTooltip = false;
     @observable showGroupsTooltip = false;
+    @observable private showModal:boolean = false;
```
Comment (alisman) [inline src/pages/studyView/StudyViewPage.tsx]: seems like this is a particular modal.  showReturnToDefaultChartList modal?

**Comment by kalletlak** [comment]
@alisman updated

**Comment by jjgao** [comment]
@kalletlak @alisman I don't think `reset charts` on the page is clear and necessary. Maybe we can rename `+ Add Chart` to `Charts` and have the `reset charts` button in the dialog?

**Pull request merged**